### PR TITLE
Update WemoLightHandler subscription handling

### LINKED ISSUE #4282: Wemo Light Handler: remove subscription
@hmerk Hi, I have a question about the current `removeSubscription` implementation.

```Java
    private synchronized void removeSubscription() {
        logger.debug("Removing WeMo GENA subscription for '{}'", this);
        if (service.isRegistered(this)) {
            String subscription = null;

            if ((subscriptionState.get(subscription) != null) && subscriptionState.get(subscription).booleanValue()) {
                logger.debug("WeMo {}: Unsubscribing from service {}...", getUDN(), subscription);
                service.removeSubscription(this, "bridge1");
            }

            subscriptionState = new HashMap<String, Boolean>();
            service.unregisterParticipant(this);

        }
    }
```

You set `subscription` to `null` and use this as a key for the `subscriptionState` map.
A key of a map is IMHO non null by default, so what do you want to check here?
What am I missing here?

### PR DESCRIPTION
**Pull request by hmerk**
Closes #4282

Signed-off-by: Hans-Jörg Merk hans-joerg.merk@t-online.de

### CODE REVIEW

File: extensions/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/handler/WemoLightHandler.java [hunk 1]
```diff
@@ -393,13 +393,13 @@ private synchronized void onSubscription() {
     }
 
     private synchronized void removeSubscription() {
-        logger.debug("Removing WeMo GENA subscription for '{}'", this);
         if (service.isRegistered(this)) {
-            String subscription = null;
+        logger.debug("Removing WeMo GENA subscription for '{}'", this);
```
Comment (triller-telekom) [inline extensions/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/handler/WemoLightHandler.java]: I think the formatter did not work correctly here, there is an indentation level missing.

File: extensions/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/handler/WemoLightHandler.java [hunk 2]
```diff
@@ -393,13 +393,13 @@ private synchronized void onSubscription() {
...
+            String subscription = "bridge1";
```
Comment (triller-telekom) [inline extensions/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/handler/WemoLightHandler.java]: While you are extracting this "magic" string, I think you should make it a constant with capital letters on top of the class, so it is easier to find.

Comment (hmerk) [reply extensions/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/handler/WemoLightHandler.java]: Sure, will change it to a constant.

File: extensions/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/handler/WemoLightHandler.java (hunk 1 above)
Comment (hmerk) [reply extensions/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/handler/WemoLightHandler.java]: Indeed, will fix it

**Comment by triller-telekom** [comment]
@hmerk: Do you have this PR still on your list? :)

**Comment by hmerk** [comment]
@triller-telekom Yes, still on my list but actually not finding time for any coding.

**Comment by hmerk** [comment]
@triller-telekom 
I finally found time to get this sorted out and already squashed my changes.
Hope it is ok now.

**Comment by triller-telekom** [comment]
@hmerk: Thank you very much!

File: extensions/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/handler/WemoLightHandler.java [hunk 3]
```diff
@@ -64,6 +64,8 @@
      */
     private static final int DIM_STEPSIZE = 5;
 
+    protected final static String SUBSCRIPTION = "bridge1";
```
Comment (kaikreuzer) [inline extensions/binding/org.eclipse.smarthome.binding.wemo/src/main/java/org/eclipse/smarthome/binding/wemo/handler/WemoLightHandler.java]: @hmerk Just for my understanding: Can you elaborate what "bridge1" here exactly means and whether this also works, if you have multiple ESH instances OR multiple WeMo devices? I see that this constant was used before your PR already, but it simply looks fishy to me...

**Comment by hmerk** [comment]
"bridge1" is the name of the eventservice we subscribe to. It's name is definde in the devices firmware, like "basicevent1" for switches or "insight1" for Insight switches.
I will create another PR next week to give the constant a more meaningfull name like EVENTSERVICE and add a short info to the code what it is about.
It has no influence to multiple devices or instances.

**Pull request merged**